src/Raven.Server/Documents/Indexes/IndexStore.cs

```diff
@@ -365,8 +369,23 @@ private void OpenIndexes()
 
                     exceptionAggregator.Execute(() =>
                     {
-                        var index = Index.Open(indexId, _documentDatabase);
-                        _indexes.Add(index);
+                        try
+                        {
+                            var index = Index.Open(indexId, _documentDatabase);
+                            _indexes.Add(index);
+                        }
+                        catch (Exception e)
+                        {
+                            // TODO arek: I think we can ignore auto indexes here, however for static ones try to retrieve names
+                            var fakeIndex = new FaultyInMemoryIndex(indexId);
```

Comment:
Yes. We don't have a name here. Is there any reason to still keep this code (creation of `FaultyInMemoryIndex`) after your changes? We won't allow to open the database anyway if any exception occurs here.